src/main/java/com/gargoylesoftware/htmlunit/javascript/host/xml/XMLHttpRequest.java

```diff
@@ -222,8 +222,14 @@ private void setState(final int state, Context context) {
         if (state == DONE) {
             final JavaScriptEngine jsEngine = (JavaScriptEngine) containingPage_.getWebClient().getJavaScriptEngine();
```

Comment:
Maybe we can forward the engine to the private method to avoid calling this several times

Reply:
Yeah sure. Seemed like simple Getters to me, that's why didn't bother adding more params to the method.